lib/honeybadger/team_filter.rb

```diff
@@ -0,0 +1,10 @@
+class TeamFilter
+  def filter(options)
+    team = Thread.current[:tn_honeybadger_team] || options[:tn_team]
```

Comment:
If you want to be able to override the current thread's honeybadger team in a specific call to `notify_rex`, you need to check `options[:tn_team]` first.

Also if you want to override vanguard over REX, you need to either have the default honeybadger API key have a non-nil team value, or check for `options.has_key?(:tn_team)`